extensions/wikia/ArticleVideo/scripts/featured-video.jwplayer.instant.js

```diff
@@ -31,27 +37,41 @@ require([
 		return !document.hidden && willAutoplay && ['playing', 'paused'].indexOf(playerInstance.getState()) === -1;
 	}
 
-	//Fallback to the generic playlist when no recommended videos playlist is set for the wiki
-	var recommendedPlaylist = videoDetails.recommendedVideoPlaylist || 'Y2RWCKuS';
+	function setupPlayer(bidParams) {
+		playerInstance.setup({
+			advertising: {
+				client: 'googima'
+			},
+			autostart: willAutoplay && !document.hidden,
+			description: videoDetails.description,
+			image: '//content.jwplatform.com/thumbs/' + videoId + '-640.jpg',
+			mute: willAutoplay,
+			playlist: videoDetails.playlist,
+			related: {
+				autoplaytimer: 5,
+				file: 'https://cdn.jwplayer.com/v2/playlists/'+ recommendedPlaylist +'?related_media_id=' + videoId,
+				oncomplete: inNextVideoAutoplayCountries ? 'autoplay' : 'show'
+			},
+			title: videoDetails.title
+		});
 
-	playerInstance.setup({
-		advertising: {
-			client: 'googima'
-		},
-		autostart: willAutoplay && !document.hidden,
-		description: videoDetails.description,
-		image: '//content.jwplatform.com/thumbs/' + videoId + '-640.jpg',
-		mute: willAutoplay,
-		playlist: videoDetails.playlist,
-		related: {
-			autoplaytimer: 5,
-			file: 'https://cdn.jwplayer.com/v2/playlists/'+ recommendedPlaylist +'?related_media_id=' + videoId,
-			oncomplete: inNextVideoAutoplayCountries ? 'autoplay' : 'show'
-		},
-		title: videoDetails.title
-	});
+		featuredVideoAds(playerInstance, bidParams);
+		featuredVideoTracking(playerInstance, willAutoplay);
+		handleTabNotActive(willAutoplay);
+	}
 
-	featuredVideoAds(playerInstance);
-	featuredVideoTracking(playerInstance, willAutoplay);
-	handleTabNotActive(willAutoplay);
+	if (a9 && adContext.get('bidders.a9Video')) {
```

Comment:
Is it possible to move logic related to `a9` into `featuredVideoAds` module?

Reply:
Unfortunately we can't add ads related event listener before instance is created